src/lm_buddy/paths.py

```diff
@@ -0,0 +1,68 @@
+from pathlib import Path
+from typing import Annotated, Any
+
+from huggingface_hub.utils import HFValidationError, validate_repo_id
+from pydantic import BeforeValidator
+
+from lm_buddy.integrations.wandb import WandbArtifactConfig
+from lm_buddy.types import BaseLMBuddyConfig
+
+
+class FilesystemPath(BaseLMBuddyConfig):
+    """Absolute path to an object on the filesystem"""
+
+    __match_args__ = ("path",)
+
+    path: Path
+
+
+class HuggingFaceRepoID(BaseLMBuddyConfig):
+    """Repository ID on the HuggingFace Hub."""
+
+    __match_args__ = ("repo_id",)
+
+    repo_id: str
+
+
+def is_valid_huggingface_repo_id(s: str):
+    """
+    Simple test to check if an HF model is valid using HuggingFace's tools.
+    Sadly, theirs throws an exception and has no return.
+
+    Args:
+        s: string to test.
+    """
+    try:
+        validate_repo_id(s)
+        return True
+    except HFValidationError:
+        return False
+
+
+def validate_loadable_path(x: Any) -> Any:
```

Comment:
suggestion `validate_path`

Reply:
Renamed to `validate_asset_path` in accordance with teh change in type name